src/lib.rs

```diff
@@ -611,6 +617,10 @@ pub trait Watcher: Sized {
     /// Returns an error in the case that `path` has not been watched or if removing the watch
     /// fails.
     fn unwatch<P: AsRef<Path>>(&mut self, path: P) -> Result<()>;
+
+    ///Sets the duration for DebouncedEvent::OnGoingWrite. When set, OnGoingWrite event will be
+    /// fired every "duration" units.
+    fn set_on_going_write_duration(&self, duration: Duration);
```

Comment:
Instead of making this a required method for the trait, make it a provided method with an empty body. Implementations that can should override it, and implementations that can't don't have to do anything.

Reply:
I agree. Made the change.